### PR TITLE
PR: Handle truncated dictionary of namespace view from the kernel

### LINKED ISSUE #4873: Error when trying to get a very large namespace view from the kernel
I keep getting the same error after running the following code:

`from brian import *`
`brian_sample_run()`

However the output seems to be fine on IPython console.

**Error**

```python-traceback
  File "/home/biomike/anaconda2/lib/python2.7/site-packages/qtconsole/base_frontend_mixin.py", line 163, in _dispatch
Traceback (most recent call last):
  File "/home/biomike/anaconda2/lib/python2.7/site-packages/qtconsole/base_frontend_mixin.py", line 163, in _dispatch
    handler(msg)
  File "/home/biomike/anaconda2/lib/python2.7/site-packages/spyder/widgets/ipythonconsole/namespacebrowser.py", line 214, in _handle_execute_reply
    self.handle_exec_method(msg)
  File "/home/biomike/anaconda2/lib/python2.7/site-packages/spyder/widgets/ipythonconsole/shell.py", line 317, in handle_exec_method
    properties = ast.literal_eval(data['text/plain'])
  File "/home/biomike/anaconda2/lib/python2.7/ast.py", line 49, in literal_eval
    node_or_string = parse(node_or_string, mode='eval')
  File "/home/biomike/anaconda2/lib/python2.7/ast.py", line 37, in parse
    return compile(source, filename, mode, PyCF_ONLY_AST)
  File "<unknown>", line 9001
    ...}
    ^
SyntaxError: invalid syntax
    handler(msg)
  File "/home/biomike/anaconda2/lib/python2.7/site-packages/spyder/widgets/ipythonconsole/namespacebrowser.py", line 214, in _handle_execute_reply
    self.handle_exec_method(msg)
  File "/home/biomike/anaconda2/lib/python2.7/site-packages/spyder/widgets/ipythonconsole/shell.py", line 311, in handle_exec_method
    view = ast.literal_eval(data['text/plain'])
  File "/home/biomike/anaconda2/lib/python2.7/ast.py", line 49, in literal_eval
    node_or_string = parse(node_or_string, mode='eval')
  File "/home/biomike/anaconda2/lib/python2.7/ast.py", line 37, in parse
    return compile(source, filename, mode, PyCF_ONLY_AST)
  File "<unknown>", line 4001
    ...}
    ^
SyntaxError: invalid syntax
```

## Version and main components

* Spyder Version: 3.2.0 
* Python Version: 2.7.13
* Qt Versions: 5.6.2, PyQt5 5.6 on Linux

## Dependencies
```
pyflakes >=0.5.0  :  1.5.0 (OK)
pycodestyle >=2.3 :  2.3.1 (OK)
pygments >=2.0    :  2.2.0 (OK)
pandas >=0.13.1   :  0.20.3 (OK)
numpy >=1.7       :  1.13.1 (OK)
sphinx >=0.6.6    :  1.6.2 (OK)
rope >=0.9.4      :  0.9.4 (OK)
jedi >=0.9.0      :  0.10.2 (OK)
psutil >=0.3      :  5.2.2 (OK)
nbconvert >=4.0   :  5.2.1 (OK)
sympy >=0.7.3     :  1.1.1 (OK)
cython >=0.21     :  0.26 (OK)
qtconsole >=4.2.0 :  4.3.0 (OK)
IPython >=4.0;<6.0:  5.3.0 (OK)
pylint >=0.25     :  1.6.4 (OK)
```

### PR DESCRIPTION
**Pull request by dalthviz**
Fixes #4873

### CODE REVIEW

**Comment by ccordoba12** [comment]
Tests in Travis are failing. Also, please add a test for this.

File: spyder/widgets/ipythonconsole/debugging.py [hunk 1]
```diff
@@ -52,10 +54,14 @@ def refresh_from_pdb(self, pdb_state):
             self.sig_pdb_step.emit(fname, lineno)
 
         if 'namespace_view' in pdb_state:
-            self.sig_namespace_view.emit(pdb_state['namespace_view'])
+            if pdb_state['namespace_view']:
```
Comment (ccordoba12) [inline spyder/widgets/ipythonconsole/debugging.py]: Why did you need to add this `if` clause here? I think we need to emit whatever namespace contents comes from the kernel.

Comment (dalthviz) [inline spyder/widgets/ipythonconsole/debugging.py]: I add this in order to pass the test `test_browse_history_dbg` since it was failing with a `malformed node or string` error, should change this for something like:

```python
            if pdb_state['namespace_view']:
                self.sig_namespace_view.emit(ast.literal_eval(
                        pdb_state['namespace_view']))
            else:
                self.sig_namespace_view.emit(
                        pdb_state['namespace_view'])
```

Comment (ccordoba12) [inline spyder/widgets/ipythonconsole/debugging.py]: Please post the error to understand why that test is failing.

File: spyder/utils/ipython/spyder_kernel.py [hunk 2]
```diff
@@ -138,7 +138,7 @@ def get_var_properties(self):
                     'array_ndim': self._get_array_ndim(value)
                 }
 
-            return properties
+            return repr(properties)
         else:
             return {}
```
Comment (ccordoba12) [inline spyder/utils/ipython/spyder_kernel.py]: Please add `repr({})` here.

File: spyder/utils/ipython/spyder_kernel.py [hunk 3]
```diff
@@ -138,9 +138,9 @@ def get_var_properties(self):
                     'array_ndim': self._get_array_ndim(value)
                 }
 
-            return properties
+            return repr(properties)
         else:
-            return {}
+            return repr({})
```
Comment (ccordoba12) [inline spyder/utils/ipython/spyder_kernel.py]: Sorry, this should be `repr(None)` to make the objects emitted when debugging the same as when users are not debugging. The last thing is done in `handle_exec_method`, here:

https://github.com/spyder-ide/spyder/blob/3.x/spyder/widgets/ipythonconsole/shell.py#L338

File: spyder/utils/ipython/spyder_kernel.py [hunk 4]
```diff
@@ -110,7 +110,7 @@ def get_namespace_view(self):
         settings = self.namespace_view_settings
         if settings:
             ns = self._get_current_namespace()
-            view = make_remote_view(ns, settings, EXCLUDED_NAMES)
+            view = repr(make_remote_view(ns, settings, EXCLUDED_NAMES))
             return view
```
Comment (ccordoba12) [inline spyder/utils/ipython/spyder_kernel.py]: Python makes this function to return `None` when there are no settings (which happens when there's no Variable Explorer in the window). That's why `test_browse_history_dbg` is failing.

That was working before (implicitly) because we were returning Python objects (in this case a dict). But it fails now because we are returning strings (through `repr`).

So we need to add here

    else:
        repr(None)
